GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCCTNanopbHelpers.m

```diff
@@ -188,6 +195,76 @@ gdt_cct_IosClientInfo GDTCCTConstructiOSClientInfo() {
   return iOSClientInfo;
 }
 
+gdt_cct_NetworkConnectionInfo GDTCCTConstructNetWorkConnectionInfo() {
+  gdt_cct_NetworkConnectionInfo networkConnectionInfo = gdt_cct_NetworkConnectionInfo_init_default;
+  if (GDTCORPlatformIsIOS()) {
+    networkConnectionInfo.network_type = GDTCCTNetworkConnectonInfoNetworkType();
+    if (networkConnectionInfo.network_type != gdt_cct_NetworkConnectionInfo_NetworkType_NONE) {
+      networkConnectionInfo.has_network_type = 1;
+      if (networkConnectionInfo.network_type == gdt_cct_NetworkConnectionInfo_NetworkType_MOBILE) {
+        networkConnectionInfo.mobile_subtype = GDTCCTNetworkConnectionInfoNetworkMobileSubtype();
+        if (networkConnectionInfo.mobile_subtype !=
+            gdt_cct_NetworkConnectionInfo_MobileSubtype_UNKNOWN_MOBILE_SUBTYPE) {
+          networkConnectionInfo.has_mobile_subtype = 1;
+        }
+      }
+    }
+  } else if (GDTCORPlatformIsMacOS() || GDTCORPlatformIsTvOS() || GDTCORPlatformIsWatchOS()) {
+    SCNetworkReachabilityFlags reachabilityFlags = [GDTCORReachability currentFlags];
+    // Currently set all non-iOS platform network type to WIFI if network is reachable.
+    if (reachabilityFlags & kSCNetworkReachabilityFlagsReachable) {
```

Comment:
if ((reachabilityFlags & kSCNetworkReachabilityFlagsReachable) == kSCNetworkReachabilityFlagsReachable)

Here's a good explanation why from http://sam.dods.co/blog/2014/02/01/checking-for-a-value-in-a-bit-mask/

Essentially, it's possible for reachabilityFlags & kSCNetworkReachabilityFlagsReachable to evaluate to YES because they have an overlapping bit, but not be equal to kSCNetworkReachabilityFlagsReachable (which requires all bits to overlap).